lib/assert.js

```diff
@@ -330,6 +330,14 @@ function getErrMessage(message, fn) {
         const { StringDecoder } = require('string_decoder');
         decoder = new StringDecoder('utf8');
       }
+
+      // ESM file prop is a file proto. Convert that to path.
+      // This ensure opensync will not throw ENOENT for ESM files.
+      const fileProtoPrefix = 'file://';
+      if (StringPrototypeStartsWith(filename, fileProtoPrefix)) {
+        filename = StringPrototypeReplace(filename, fileProtoPrefix, '');
```

Comment:
Please use `fileURLToPath`. Removing the prefix doesn't work in all cases, especially on Windows.

Reply:
fileToUrlpath is only handles the URL. It was my initial thought, then moved to this approach and decided to test in win32 separately.

Else, we need to parse to URL first and convert back to path to use that method.

Reply:
This is an error case. I think it's not a problem to parse the URL.

Reply:
Okay. Will make the changes.

Reply:
Done. Please check.